connector/src/main/java/org/geysermc/connector/network/translators/java/window/JavaSetSlotTranslator.java

```diff
@@ -86,8 +89,24 @@ public void translate(GeyserSession session, ServerSetSlotPacket packet) {
                 session.getPlayerInventory().setItem(packet.getSlot(), newItem, session);
                 InventoryTranslator.PLAYER_INVENTORY_TRANSLATOR.updateSlot(session, session.getPlayerInventory(), packet.getSlot());
             } else {
-                inventory.setItem(packet.getSlot(), newItem, session);
-                translator.updateSlot(session, inventory, packet.getSlot());
+                if(packet.getWindowId() == 0 && packet.getSlot() == 5) {
+                    CompoundTag profile = null;
+
+                    if (session.getItemMappings().getMapping(newItem.getItemData(session)).getJavaIdentifier().equals("minecraft:player_head")
+                            && newItem.getNbt() != null
+                            && newItem.getNbt().contains("SkullOwner")
+                            && newItem.getNbt().get("SkullOwner") instanceof CompoundTag) {
+                        profile = newItem.getNbt().get("SkullOwner");
```

Comment:
The same suggestion can be applied here.

Reply:
![Bildschirmfoto 2021-11-17 um 09 38 35](https://user-images.githubusercontent.com/20094418/142165936-77c0d035-37de-448f-bebb-1ac34d81662d.png)
This doesn't make sense, does it?

Reply:
You can move the if statement with setOwnFakeHead down below into this if statement.

Reply:
Can't because it should even be called when there is no skull owner or it isn't a player head

Reply:
Ah, my bad.